src/quote/zeroEx/componentsQuoteProvider.ts

```diff
@@ -0,0 +1,112 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { ZeroExApi } from '../../utils/0x'
+import { QuoteToken } from '../quoteToken'
+
+export type ComponentQuotesResult = {
+  componentQuotes: string[]
+  inputOutputTokenAmount: BigNumber
+}
+export class ComponentsQuoteProvider {
+  constructor(
+    readonly chainId: number,
+    readonly slippage: number,
+    readonly wethAddress: string,
+    readonly zeroExApi: ZeroExApi
+  ) {}
+
+  /**
+   * Returns the component quotes and input/output token amount for given components
+   * and positions.
+   *
+   * @param components      An array of components
+   * @param positions       An array of positions
+   * @param isMinting       Minting or redeeming
+   * @param inputToken      The input token
+   * @param outputToken     The output token
+   *
+   * @returns a ComponentQuotesResult
+   */
+  async getComponentQuotes(
+    components: string[],
+    positions: BigNumber[],
+    isMinting: boolean,
+    inputToken: QuoteToken,
+    outputToken: QuoteToken
+  ): Promise<ComponentQuotesResult | null> {
+    if (components.length === 0 || positions.length === 0) return null
+    if (components.length !== positions.length) return null
+
+    const { chainId, slippage, zeroExApi } = this
+
+    const inputTokenAddress = this.getTokenAddressOrWeth(inputToken)
+    const outputTokenAddress = this.getTokenAddressOrWeth(outputToken)
+
+    // 0xAPI expects percentage as value between 0-1 e.g. 5% -> 0.05
+    const slippagePercentage = slippage / 100
+
+    const quotePromises: Promise<any>[] = []
+
+    components.forEach((component, index) => {
+      const buyAmount = positions[index]
+      const sellAmount = positions[index]
```

Comment:
I was thinking, here I should just be able to do something like `amount = positions[index]` right? Because it's always the same no matter if buy/sell?